common/util.go

```diff
@@ -257,7 +257,8 @@ func IsServiceTransientError(err error) bool {
 	case errors.As(err, &typesInternalServiceError):
 		return true
 	case errors.As(err, &typesServiceBusyError):
-		return true
+		// If the service busy error is due to workflow id rate limiting, then it is not transient
+		return typesServiceBusyError.Reason != WorkflowIDRateLimitReason
```

Comment:
Please add tests to cover new lines in mappers and this file

Reply:
I think *is a transient* error? it's just requiring backoff, though in the case of frontend it doesn't make sense to retry. 

If I'm not mistaken, what's missing from this classifying function is context as to where it's being used - error handling for frontend and history will probably need different definitions of what is retriable. I worry that with this rate-limit being potentially added by various non-frontend places with @sankari165's work it will possibly lead to a lot of unnecessarily failed tasks, DLQ'd messages etc. 

May I suggest either we split this  `IsServiceTransientError` into a function specific to frontend vs background processing either here or when Sankari's implementation lands

Reply:
Coverage of mappers added in this PR: https://github.com/uber/cadence/pull/5795